orcoursetrion/tests/test_github.py

```diff
@@ -337,6 +341,132 @@ def test_lib_put_team_membership_fail(self):
+        git.config('user.name', config.ORC_GH_NAME)
+        for commit in [commit_1, commit_2]:
+            with open('test', 'w') as test_file:
+                test_file.write(commit)
+            git.add('.')
+            git.commit(m=commit_2)
+
+        # Also create a branch to test branches
+        git.checkout('-b', branch)
+        with open('test', 'w') as test_file:
+            test_file.write(branch)
+        git.add('.')
+        git.commit(m=branch)
+
+        git.checkout('master')
+
+        # Now create the initial bare repo at the destination
+        sh.cd(tmp_dir_dst)
+        sh.mkdir(DST_REPO)
+        sh.cd(DST_REPO)
+        sh.git.init(bare=True)
+
+        # Alright, run the copy and verify the expected commit
+        # message, and that the test file has the last commit string
+        git_hub = GitHub(self.URL, self.OAUTH2_TOKEN)
+        src_repo_url = 'file://{0}'.format(os.path.join(tmp_dir_src, SRC_REPO))
+        dst_repo_url = 'file://{0}'.format(os.path.join(tmp_dir_dst, DST_REPO))
+
+        git_hub.shallow_copy_repo(
+            src_repo_url,
+            dst_repo_url,
+            {'email': config.ORC_GH_EMAIL, 'name': config.ORC_GH_NAME}
+        )
+
+        # Verify things are looking right.
+        sh.cd(tmp_dir_dst)
+        # Clone bare destination repo and verify contents
+        DST_REPO_CLONE = 'cloned'
+        sh.git.clone(dst_repo_url, DST_REPO_CLONE)
+        sh.cd(DST_REPO_CLONE)
+
+        # Assert file is as expected
+        with open('test', 'r') as test_file:
+            self.assertEqual(commit_2, test_file.read())
+        git_log = sh.git.log(
+            '-1', format='%s', _tty_out=False
+        ).rstrip('\n')
+
+        # Assert commit message is correct
+        self.assertEqual(
+            git_log,
+            'Initial rerun copy by Orcoursetrion from {0}'.format(src_repo_url)
+        )
+
+        # Assert author is correct
+        git_log = sh.git.log(
+            '-1', format='%cn', _tty_out=False
+        ).rstrip('\n')
+        self.assertEqual(git_log, config.ORC_GH_NAME)
+        git_log = sh.git.log(
+            '-1', format='%ce', _tty_out=False
+        ).rstrip('\n')
+        self.assertEqual(git_log, config.ORC_GH_EMAIL)
+
+        # Delete old clone and run it with a different branch
+        sh.cd(tmp_dir_dst)
+        shutil.rmtree(DST_REPO_CLONE)
+        git_hub.shallow_copy_repo(
+            src_repo_url,
+            dst_repo_url,
+            {'email': config.ORC_GH_EMAIL, 'name': config.ORC_GH_NAME},
+            branch=branch
+        )
+
+        # Verify things are looking right in the branch clone
+        sh.cd(tmp_dir_dst)
+        # Clone bare destination repo and verify contents
+        DST_REPO_CLONE = 'cloned'
+        sh.git.clone(dst_repo_url, DST_REPO_CLONE)
+        sh.cd(DST_REPO_CLONE)
+
+        # Assert file is as expected
+        with open('test', 'r') as test_file:
+            self.assertEqual(branch, test_file.read())
```

Comment:
Agreed, I could split it into `test_branch_copy` and `test_copy` or similar, but it would slow down tests by a second or two to do the setup twice, and the setup is the same for both.